injective_functions/exchange/exchange.py

```diff
@@ -19,29 +19,28 @@ def __init__(self, chain_client) -> None:
         super().__init__(chain_client)
 
     async def get_subaccount_deposits(
-        self, subaccount_idx: int, denoms: str = None
+        self, subaccount_idx: int, denoms: List[str] = None
     ) -> Dict:
         try:
 
-            subaccount_id = await self.chain_client.address.get_subaccount_id(
-                subaccount_idx
-            )
+            subaccount_id = self.chain_client.address.get_subaccount_id(subaccount_idx)
             deposits = await self.chain_client.client.fetch_subaccount_deposits(
                 subaccount_id=subaccount_id
-            )["deposits"]
+            )
+            deposits = deposits["deposits"]
             denom_decimals = await fetch_decimal_denoms(self.chain_client.network_type)
             human_readable_deposits = {}
-            if len(denoms) > 0:
+            if denoms:
```

Comment:
This block looks somewhat difficult to parse, this logic is way more cleaner: 

```
     # Determine target denoms (all if not specified)
        target_denoms = denoms or deposits.keys()

        for denom in target_denoms:
            if denom in deposits and denom in denom_decimals:
                # Convert balances to human-readable format
                available_balance = int(deposits[denom]["availableBalance"]) / 10 ** denom_decimals[denom]
                total_balance = int(deposits[denom]["totalBalance"]) / 10 ** denom_decimals[denom]
                human_readable_deposits[denom] = {
                    "available_balance": f"{available_balance:.6f}",
                    "total_balance": f"{total_balance:.6f}",
                }
            else:
                # Handle any missing balances
                human_readable_deposits[denom] = {
                    "available_balance": "balance not found",
                    "total_balance": "balance not found",
                }
```

Reply:
added comments and refactor